src/main/java/com/flightstats/hub/spoke/SpokeTtlEnforcer.java

```diff
@@ -74,46 +65,47 @@ public SpokeTtlEnforcer(SpokeStore spokeStore,
     }
 
     private long removeFromChannelByTime(String channelPath, String timePath) {
-        String path = channelPath + "/" + timePath;
-        int waitTimeSeconds = 3;
+        final String path = channelPath + "/" + timePath;
+        final int waitTimeSeconds = 3;
         return FileUtils.deleteFiles(path, waitTimeSeconds);
     }
 
     private void updateOldestItemMetric() {
-        Optional<ChannelContentKey> potentialItem = spokeContentDao.getOldestItem(spokeStore);
-        long oldestItemAgeMS = potentialItem.isPresent() ? potentialItem.get().getAgeMS() : 0;
+        final Optional<ChannelContentKey> potentialItem = spokeContentDao.getOldestItem(spokeStore);
+        final long oldestItemAgeMS = potentialItem.map(ChannelContentKey::getAgeMS).orElse(0L);
         statsdReporter.gauge(buildMetricName("age", "oldest"), oldestItemAgeMS);
     }
 
     private String buildMetricName(String... elements) {
-        String prefix = String.format("spoke.%s", spokeStore);
-        Stream<String> stream = Stream.concat(Stream.of(prefix), Arrays.stream(elements));
+        final String prefix = String.format("spoke.%s", spokeStore);
+        final Stream<String> stream = Stream.concat(Stream.of(prefix), Arrays.stream(elements));
         return stream.collect(Collectors.joining("."));
     }
 
-    private class SpokeTtlEnforcerService extends AbstractScheduledService {
-        @Override
-        protected void runOneIteration() throws Exception {
-            try {
-                long start = System.currentTimeMillis();
-                AtomicLong evictionCounter = new AtomicLong(0);
-                log.info("running ttl cleanup");
-                TtlEnforcer.enforce(storagePath, channelService, handleCleanup(evictionCounter));
-                updateOldestItemMetric();
-                statsdReporter.gauge(buildMetricName("evicted"), evictionCounter.get());
-                long runtime = (System.currentTimeMillis() - start);
-                log.info("completed ttl cleanup {}", runtime);
-                statsdReporter.gauge(buildMetricName("ttl", "enforcer", "runtime"), runtime);
-            } catch (Exception e) {
-                log.info("issue cleaning up spoke", e);
-            }
-        }
+    private void initSpokeStore(SpokeStore spokeStore) {
+        this.spokeStore = spokeStore;
+        this.storagePath = spokeProperties.getPath(spokeStore);
+        this.ttlMinutes = spokeProperties.getTtlMinutes(spokeStore) + 1;
+    }
 
-        @Override
-        protected Scheduler scheduler() {
-            return Scheduler.newFixedRateSchedule(1, 1, TimeUnit.MINUTES);
-        }
+    void cleanup(SpokeStore spokeStore) {
+        try {
+            initSpokeStore(spokeStore);
 
-    }
+            final long start = System.currentTimeMillis();
```

Comment:
How do you feel about moving the execution time logging and exception handling out to the abstract scheduled service? That way the only responsibility of this method would be to clean up and log how many items were evicted. If you agree, you don't necessarily need to change it now..